mmcv/cnn/bricks/conv_module.py

```diff
@@ -147,6 +144,12 @@ def __init__(self,
                 norm_channels = in_channels
             self.norm_name, norm = build_norm_layer(norm_cfg, norm_channels)
             self.add_module(self.norm_name, norm)
+            if self.with_bias:
+                batch_norm_list = (nn.BatchNorm1d, nn.BatchNorm2d,
+                                   nn.BatchNorm3d)
+                if issubclass(norm, batch_norm_list):
```

Comment:
```python
from mmcv.utils import TORCH_VERSION, _BatchNorm
if issubclass(norm, _BatchNorm):
```

Reply:
I noticed that SyncBatchNorm is also inherited from _BatchNorm, I guess the bias is applied to minibatch, while the norm applies to the bigger batch, so ?

Reply:
I guess the bias has no effect on batchnorm and at the same time the synchronization of batchnorm has not relation with bias.

Reply:
Here is my point: we know that after conv, you have options to use bias inside the conv or not, which is the last operation of conv module. 
Next, if you use BatchNorm, the previous bias in conv would be cancelled out during norm operation. So we want to give warnings that these kinds of BN and conv bias combination is unnecessary.
As for GN or SyncBatchNorm or others, you can't cancel out the conv bias. For GN, the conv bias is applied to a batch, but norms are applied separately to feature groups. Similarly, for SyncBatchNorm, conv bias is applied to local minibatch, while norm is based on the sum of all mini batches. I guess the the bias inside conv would affect the outputs after SyncBatchNorm. 
Back to my point, I think the warning message is intent to remind us, "Hey, mate, you are doing some meaningless operations" instead of "Woops, you are using bias in conv, which followed by batch norm"

Reply:
We can see the implementation of SyncBN here(https://github.com/chrisway613/Synchronized-BatchNormalization/blob/master/sync_bn.py#L55).  Master rank will collect the sum and square-sum from other ranks . Master rank first calculates the mean and variance and then broadcast to others. Anything is the same as BN, so the bias has no effect on SycnBN. BTW, If I have a misunderstanding of SycnBN, please let me know

Reply:
Yeah, I am not very familiar with SycnBN, I will check the resource you mentioned, and perhaps do an experiment. Thank you.

Reply:
You are right. I was wrong to think every copy of the model on different GPUS are different from each other, and averaged at the end.